src/Language/Fixpoint/Solver/PLE.hs

```diff
@@ -875,13 +921,61 @@ evalApp γ ctx e0 es et
         else do
           return (Nothing, noExpand)
 
+-- **ETA**: PLE before the patch to perform eta expansion was only able to unfold
+-- fully applied functions, by eta equivalence we mean the property that for all
+-- functions f the terms f and (\x -> f x) have the same semantics, we leverage 
+-- this fact to fully apply functions with extra function arguments so to trigger
+-- the standard PLE expansion, see the following example:
+-- We have a function f:
+-- define f (x : int) : int = {(x)}
+-- And we need to prove some constraint of this shape
+-- { f = something }
+-- The original PLE cannot do anything on f as it's not fully applied, instead
+-- if we perform eta expansion on f we obtain the following constraint
+-- { \y:Int -> f y = something }
+-- And now PLE can perform the unfolding of f as it's now fully saturated
+-- { \y:Int -> y = something }
+-- Clearly we need the higerorder flag active as we are generating lambda in
+-- the constraints.
+evalApp _γ _ctx e0 es _et
+  | ECst (EVar _f) sortAnnotation@FFunc{} <- e0
```

Comment:
I just noticed this is relying on `ECst` while the earlier equations look for the function defition in `γ`. What is the reason to not do the same here?

Reply:
I've added a comment in the code